pkg/operator/apis/monitoring/v1alpha1/types.go

```diff
@@ -373,18 +344,82 @@ func (pm *PodMonitoring) endpointScrapeConfig(index int) (*promconfig.ScrapeConf
 		},
 	}
 
-	// Filter targets that belong to selected services.
+	// Filter targets that belong to selected pods.
+	selectors, err := relabelingsForSelector(pm.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
+	relabelCfgs = append(relabelCfgs, selectors...)
+
+	metadataLabels := map[string]struct{}{}
```

Comment:
Does it make sense to capture this logic in its own function and re-use for PodMonitoring and ClusterPodMonitoring? e.g.
```go
func relabelingsForMetadata(labels ...string) []*relabel.Config  {
	// For loop over labels and use a switch statement to capture and create each rule.
}
```

And the checking the set of the metadata list can be done in each type before the function call, e.g. for PodMonitoring
```go
if metadata := *pm.Spec.TargetLabels.Metadata; metadata != nil {
	for _, l := range metadata {
		if allowed := []string{"pod", "container", "node"}; !containsString(allowed, l) {
			return nil, errors.Errorf("metadata label %q not allowed, must be one of %v", l, allowed)
		}
	}
	metadataRelabelings = relabelingsFromMetadata(metadata...)
	relabelCfgs = append(relabelCfgs, metadataRelabelings...)
}
```
Something similar would follow for ClusterPodMonitoring.